src/java/picard/illumina/MarkIlluminaAdapters.java

```diff
@@ -224,8 +226,8 @@ protected int doWork() {
                     progress.record(r);
                     if (out != null) out.addAlignment(r);
 
-                    final Integer clip = rec.getIntegerAttribute(ReservedTagConstants.XT);
-                    if (clip != null) histo.increment(rec.getReadLength() - clip + 1);
+                    final Integer clip = r.getIntegerAttribute(ReservedTagConstants.XT);
+                    if (clip != null) histo.increment(r.getReadLength() - clip + 1);
```

Comment:
This doesn't matter so much in the cases where the reads are of the same length, since the `XT` tags will be set to the same value for both ends.  This should only affect the cases where one end is longer than the other, which happens infrequently by design for Illumina data.